dags/common/enricher.py

```diff
@@ -0,0 +1,95 @@
+import xml.etree.ElementTree as ET
+from logging import Logger
+from typing import Dict
+
+import backoff
+import requests
+from structlog import get_logger
+
+
+class Enricher(object):
+    def __init__(self) -> None:
+        self.logger = get_logger().bind(class_name=type(self).__name__)
+
+    def _get_schema(self):
+        return "http://repo.scoap3.org/schemas/hep.json"
+
+    @backoff.on_exception(
+        backoff.expo, requests.exceptions.RequestException, max_time=60, max_tries=3
+    )
+    def _get_arxiv_categories(self, arxiv_id=None):
+        if arxiv_id is None:
+            raise ValueError("Arxiv is None.")
+
+        def clean_arxiv(arxiv):
+            if arxiv is None:
+                return None
+
+            return arxiv.split(":")[-1].split("v")[0].split(" ")[0].strip("\"'")
+
+        def get_arxiv_categories_from_response_xml(xml: ET.Element):
+            xml_namespaces = {
+                "arxiv": "http://arxiv.org/schemas/atom",
+                "w3": "http://www.w3.org/2005/Atom",
+            }
+
+            entries = xml.findall("./w3:entry", namespaces=xml_namespaces)
+            if len(entries) != 1:
+                return []
+            entry = entries[0]
```

Comment:
what if `len(entries) == 0`?

Reply:
Then return an empty array

Reply:
no this will fail, you are checking just for `len(entries) != 1` which means the `len(entries) ==  0` will go to next line and call `entry = entries[0]` which will fail.

```python
In [2]: categories = []

In [3]: len(categories)
Out[3]: 0

In [4]: categories[0]
---------------------------------------------------------------------------
IndexError                                Traceback (most recent call last)
<ipython-input-4-56c311044c1c> in <module>()
----> 1 categories[0]

IndexError: list index out of range
```

Reply:
As discussed, the code returns an empty array in case of `len(entries) != 1`